bin/desi_link_calibnight

```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env python
+
+import sys
+from desispec.scripts import link_calibnight
+
+sys.exit(link_calibnight.main())
```

Comment:
Context: I originally implemented this as returning an integer and having bin/desi_link_calibnight wrap that with a sys.exit(main()).  But then I realized that could be dangerous if desispec.scripts.link_calibnight.main() was called directly as part of a pipeline that didn't know to treat non-zero integers as bad, so I wanted something that would fail more dramatically so switched to sys.exit(errcode).

Prompted by this question, I checked how we've been doing it elsewhere, and we haven't been consistent.  e.g. scripts.preproc also calls sys.exit, while scripts.extract raises exceptions, while scripts.zproc does some of both.  In the end I think either way is fine, and I checked that sys.exit(sys.exit(7)) results in a unix error code of 7, so I'm going to leave this as is.  If we find a compelling reason why sys.exit(sys.exit(n)) is bad, we'll need to cleanup more than just this script.